webapi/src/main/scala/org/knora/webapi/routing/v1/ResourcesRouteV1.scala

```diff
@@ -388,8 +406,11 @@ object ResourcesRouteV1 extends Authenticator {
                             )
                     }.toSet -- intermediateResults.keySet - initialOntologyIri
 
+                    // Make a set of all the ontologies referenced by the initial ontology.
+                    referencedOntologies: Set[IRI] = ontologyIrisFromCardinalities ++ ontologyIrisFromObjectClassConstraints
+
                     // Recursively get NamedGraphEntityInfoV1 instances for each of those ontologies.
-                    futuresOfNamedGraphInfosFromObjectClassConstraints: Set[Future[Map[IRI, NamedGraphEntityInfoV1]]] = ontologyIrisFromObjectClassConstraints.map {
+                    futuresOfNamedGraphInfosForReferencedOntologies: Set[Future[Map[IRI, NamedGraphEntityInfoV1]]] = referencedOntologies.map {
                         ontologyIri =>
                             getNamedGraphInfosRec(
```

Comment:
Is it guaranteed that the recursive call of `getNamedGraphInfosRec` terminates? What about ontologies that cross reference each others' definitions? E.g., an ontology A that has a resource class with a cardinality for a property defined in B, and vice versa.

Reply:
It collects the named graphs that it's already traversed, so it doesn't traverse them again.

Reply:
But if you want a guarantee, you'll have to solve this problem first :)

https://en.m.wikipedia.org/wiki/Halting_problem

Reply:
a) I think we should be pragmatic about this
b) I think You should write a paper about it

Reply:
Could we look at it once you arrive here in the office?

Reply:
a) Yes.
b) No.
c) Yes, let's look at it together. :)

Reply:
c) ok, see you later!